unittests/savanna_misc_tests.cpp

```diff
@@ -681,10 +799,30 @@ BOOST_FIXTURE_TEST_CASE(verify_spring_1_0_block_compatibitity, savanna_cluster::
    BOOST_REQUIRE_EQUAL(qc_s(qc(b9)), strong_qc(b8));    // b9 claims a strong QC on b8
    BOOST_REQUIRE_EQUAL(A.lib_number, b6->block_num());  // b9 makes B6 final
 
-   // check that the block id of b9 match what we got with Spring 1.0.0
+   std::filesystem::path test_data_path { UNITTEST_TEST_DATA_DIR };
+   auto ref_blockchain_path = test_data_path / "consensus_blockchain";
+
+   // check that the block id of b9 match what we got before.
    auto b9_id = b9->calculate_id();
-   BOOST_REQUIRE_EQUAL(b9_id, block_id_type{"00000013725f3d79bd4dd4091d0853d010a320f95240981711a942673ad87918"});
+   BOOST_REQUIRE_EQUAL(b9_id, read_reference_id(ref_blockchain_path));
+
+   if (save_blockchain) {
+      save_blockchain_data(ref_blockchain_path, A.get_config().blocks_dir, b9_id, snapshot);
+      return;
+   }
+
+   block_log blog(ref_blockchain_path);
+
+   // replay the reference blockchain and make sure LIB id in the replayed
+   // chain matches reference LIB id
+   // --------------------------------------------------------------------
+   fc::temp_directory temp_dir; // need to pass in temp_dir. otherwise it would be destroyed after replay_reference_blockchain returns
+   std::unique_ptr<tester> replay_chain = replay_reference_blockchain(ref_blockchain_path, temp_dir, blog);
 
+   // start another blockchain using reference snapshot, and sync with the blocks
+   // from the replayed blockchain
+   // ---------------------------------------------------------------------------
+   sync_replayed_blockchain(ref_blockchain_path, std::move(replay_chain), blog);
```

Comment:
This should check that the result of replaying the blockchain matches `b1` through `b9`. Unless I'm mistaken, here it just checks the consistency of what is stored in the reference chain.

Reply:
At the end of both `replay_reference_blockchain()` and `sync_replayed_blockchain()`, replayed/synched block ID is verified against with reference block ID